### PR TITLE
Added chipsets to supported platforms

### LINKED ISSUE #6713: Align supported platforms for Flutter & Dart
Create & document a process to proactively approach platform misalignments and supported platforms for Flutter & Dart. This is a child issue of master issue flutter/website#6710

### LINKED ISSUE #10136: Update supported platforms to show supported architectures
### Describe the problem

The [supported deployment platforms](https://docs.flutter.dev/reference/supported-platforms) page does not specify which architectures are supported.

### Expected fix

Show supported platforms and architectures.

Example: Windows Arm64, X86 etc.

### On which browser(s) did you experience this issue?

_No response_

### Additional context

_No response_

### I would like to fix this problem.

- [ ] I will try and fix this problem on docs.flutter.dev.

### PR DESCRIPTION
**Pull request by atsansone**
Added hardware architectures to supported platforms. 

Fixes #10136
Fixes #6713

### CODE REVIEW

**Review by sfshaza2 — APPROVED** [review]
Assuming the chipset info is correct, lgtm

File: src/reference/supported-platforms.md
Comment (parlough) [inline src/reference/supported-platforms.md:31]: Flutter has no `dev` channel, only `beta` and `main`.

File: src/_data/platforms.yml
Comment (parlough) [inline src/_data/platforms.yml:35]:
```suggestion
  besteffort: '15.6 or later'
```

File: src/_data/chipsets.yml
Comment (parlough) [inline src/_data/chipsets.yml:6]: This is probably better off as the "N/A" icon. Windows for RISC-V simply doesn't exist right now.

**Review by parlough — COMMENTED** [review]
Thanks for working on this @atsansone! Cool idea!

A few accessibility and accuracy issues I noticed during an initial review:

Let me know if want to discuss any further. Thanks!

File: src/reference/supported-platforms.md
Comment (sfshaza2) [reply src/reference/supported-platforms.md:31]: Good catch! @parlough is indeed correct!

**Comment by sfshaza2** [comment]
Thanks so much for your thorough review, @parlough!

**Review by domesticmouse — COMMENTED** [review]
LGTM once @parlough's changes are landed

File: src/reference/supported-platforms.md
Comment (atsansone) [reply src/reference/supported-platforms.md:22]: We did, but not anymore. This table came from the Dart site.

Comment (loic-sharma) [inline src/reference/supported-platforms.md:22]: FYI, we were told by Microsoft to use the "Windows Arm64" branding instead of "Windows ARM64". It looks like that's the branding used on arm.com as well ([link](https://www.arm.com/architecture/cpu)). Should we rebrand these to `Arm32` and `Arm64`?

Comment (loic-sharma) [inline src/reference/supported-platforms.md:22]: Should we upgrade Windows Arm64 to "Supported in all channels"? The 3.19 blog post declared "initial support" for Windows Arm64 on beta and master channels: https://medium.com/flutter/whats-new-in-flutter-3-19-58b1aae242d2

FYI we're in a weird messy state right now: you can create Windows Arm64 executables _if you re-download the engine artifacts once_ (examples: `flutter precache`, `flutter upgrade`, etc..). Until you re-download artifacts, you'll create Windows x64 executables that run on Windows Arm64 using emulation.

Some context: On Windows, we don't support cross-compilation yet and the Flutter tool targets whatever architecture the tool was built for. If on a Windows Arm64 machine you use the Flutter tool for x64 machines, you'll produce an x64 app. Although we do have Windows Arm64 artifacts, we haven't created a Windows Arm64 Flutter SDK .zip yet. As a result, Windows Arm64 users start with x64 artifacts until they re-download their artifacts.

Sorry for the long-winded explanation.

File: src/reference/supported-platforms.md
Comment (atsansone) [reply src/reference/supported-platforms.md]: I found a way to handle this. Take a look.

File: src/reference/supported-platforms.md
Comment (atsansone) [reply src/reference/supported-platforms.md]: Fixed.

Comment (atsansone) [reply src/reference/supported-platforms.md]: Sounds good!

Comment (atsansone) [reply src/reference/supported-platforms.md]: Understood and fixed.

File: src/reference/supported-platforms.md
Comment (atsansone) [reply src/reference/supported-platforms.md]: Fixed.

File: src/_data/chipsets.yml
Comment (atsansone) [reply src/_data/chipsets.yml]: See @loic-sharma comment [below](https://github.com/flutter/website/pull/10174/files#r1498002857).

File: src/reference/supported-platforms.md
Comment (parlough) [reply src/reference/supported-platforms.md]: Is there any reason in particular you made the switch? Can we switch back? In the SDK and the GCP storage archive, it is labeled as IA32, so I think it's best to be consistent across everywhere if possible.

File: src/reference/supported-platforms.md
Comment (loic-sharma) [inline src/reference/supported-platforms.md:32]: Windows Arm64 is "initially supported" on all channels

**Review by loic-sharma — APPROVED** [review]

File: src/reference/supported-platforms.md
Comment (parlough) [inline src/reference/supported-platforms.md:43]:
```suggestion
| Platform |   IA32 (x86)   |   x64   |  ARM32  |   ARM64   | RV64GC   |
```

File: src/reference/supported-platforms.md
Comment (parlough) [inline src/reference/supported-platforms.md:55]: - For more reliable and easier to notice forced new-lines, you can use `<br>` here. We do it elsewhere on the sites as well.
- Generally I think of an OS or architecture as "supported on" or "supported by".
- To highlight channels, we generally use code font or bold while keeping it lower case.
- "No version exists" could be mistaken to mean that no version exists for Flutter. I think it would be beneficial to expand a bit to different it from the case where it's just unsupported.

```suggestion
{{yes}} Supported on all channels.<br>
{{no}} Unsupported on all channels.<br>
{{beta}} Supported on the `beta` and `main` channels only.<br>
{{na}} Unsupported by the operating system.<br>
```

File: src/reference/supported-platforms.md
Comment (parlough) [inline src/reference/supported-platforms.md:63]: Consider using a definition list here instead, since it seems to fit your use case and adds extra semantics.

File: src/reference/supported-platforms.md
Comment (parlough) [inline src/reference/supported-platforms.md:66]: I think it's better to not drag "Google" in to all three descriptions. Perhaps it makes sense for the "Supported" category, but the other two are likely better off as "The Flutter team" or "Flutter".

For the **supported** category, I think it also should be "The Flutter team" but can perhaps be strengthened with a mention of Google. I'm not sure exactly how yet, so I'll try to come back to this comment with ideas. Perhaps something like "The Flutter team, supported by Google, runs tests for these platforms on every commit."?

**Review by parlough — COMMENTED** [review]
Thanks for those adjustments Tony! I haven't had a chance to take a look at the a11y improvements to the icons, but thanks for the additional focus there.

Some comments as I was reading through again:

File: src/reference/supported-platforms.md
Comment (atsansone) [reply src/reference/supported-platforms.md:63]: I tried that, but it looked like it added a bit too much page length. Let me check again.

File: src/reference/supported-platforms.md
Comment (atsansone) [reply src/reference/supported-platforms.md:66]: That's fair. The original said Google; I was just updating.

File: src/reference/supported-platforms.md
Comment (atsansone) [reply src/reference/supported-platforms.md:55]: @parlough : No issue with most of this, but the N/A comes from mit's thoughts on Dart. I don't know if I want to confuse this if you look across both sites.

File: src/reference/supported-platforms.md
Comment (atsansone) [reply src/reference/supported-platforms.md:66]: Fixed.

File: src/reference/supported-platforms.md
Comment (atsansone) [reply src/reference/supported-platforms.md:55]: I would also like to keep markup to Markdown wherever possible.

File: src/_data/chipsets.yml
Comment (zanderso) [inline src/_data/chipsets.yml:28]: This is definitely not correct. 32-bit ARM Android is 100% supported and we have the devices in our lab.

**Review by zanderso — COMMENTED** [review]
I have found at least one mistake on this page. Can we please revert this change and have it reviewed by more team members?

File: src/_data/chipsets.yml
Comment (atsansone) [reply src/_data/chipsets.yml:28]: We can do another PR. Is there a definitive lust, @zanderso?

Comment (zanderso) [reply src/_data/chipsets.yml:28]: I would suggest adding the TLs from the teams that support each platform for the definitive answers.

**Pull request merged**
